### PR TITLE
Export stuff that might be needed elsewhere

### PR DESCRIPTION
**Pull request by askvortsov1**
The necessity for this came about while working on https://github.com/FriendsOfFlarum/drafts/pull/10.

I am not sure that this is the correct way to accomplish this.

### CODE REVIEW

**Comment by datitisev** [comment]
You have to use `Object.assign` for `flarum.core.compat`

https://github.com/flarum/tags/blob/b2f74cbb9020f474e1659aba6ae3f1e408742216/js/src/forum/index.js#L36-L40

**Comment by askvortsov1** [comment]
That's the first thing I tried, but wasnt sure we wanted to do it that way: wouldn't that add it to the flarum namespace vs an fof namespace?

**Comment by datitisev** [comment]
There's no standard here to uphold. If you want to keep them separate, do not export it as `compat` - simply export everything in `compat` itself. We do not need this compat object then.

**Comment by clarkwinkelmann** [comment]
The core compat object is meant for core extensions only.

I think the approach here is the correct approach, but I don't like the fact the file is named `compat` when it has nothing to do with the compat feature of Flarum core.

What I have done in my Formulaire extension to export everything while keeping the code tidy is:

`index.js`:

```js
export * from './components';
export * from './pages';

app.initializers.add('kilowhat-formulaire', () => {
    //
}
```

`components/index.js`:

```js
import ErrorPage from './ErrorPage';
import FormGroup from './FormGroup';
import FormTitle from './FormTitle';
import LoadingIndicator from './LoadingIndicator';

export const components = {
    ErrorPage,
    FormGroup,
    FormTitle,
    LoadingIndicator,
};
```

And repeat with an `index.js` file in each directory.

But this means the exports look like

```
{
    components: {
        TheComponent: TheComponent
    }
}
```

instead of

```
{
      'components/TheComponent': TheComponent
}
```

So I guess we can't import it with the `@vendor/namespace/component` syntax, but that syntax is broken already anyway I think.

**Comment by askvortsov1** [comment]
Yeah that syntax is unfortunately broken. It would be nice to figure out how to get extensions to export stuff automatically (if that's even a possibility), but thats not really a priority in the release cycle rn. I'll change it to export an object of components, and an object of helpers?

**Comment by datitisev** [comment]
> It would be nice to figure out how to get extensions to export stuff automatically (if that's even a possibility)

I think I've looked at this, but I either didn't know what to look up or didn't find anything afaik. 😕 

> I'll change it to export an object of components, and an object of helpers?

Sounds good.

**Comment by askvortsov1** [comment]
@datitisev is this ok?

File: js/src/forum/index.js [hunk 1]
```diff
@@ -79,3 +79,5 @@ app.initializers.add('fof-byobu', function (app) {
         });
     });
 });
+
+export * from './compat';
```
Comment (datitisev) [inline js/src/forum/index.js:83]: These lines are no longer needed.

Comment (datitisev) [reply js/src/forum/index.js:83]: Never mind, I am being an idiot.

**Comment by datitisev** [comment]
I think it's better if we do what Clark does in his formulaire extension, instead of having a whole file dedicated to it (like core does).

**Comment by askvortsov1** [comment]
My thought was avoiding clogging index.js with ALL the imports.

**Comment by datitisev** [comment]
Well, that's not what Clark said. Clark has a `components/index.js` - we can do the same. Have `components/index.js` and `helpers/index.js` (can be required as `./components` and `./helpers`), and export `*` from them.

**Comment by askvortsov1** [comment]
Ah my bad. I'll have this fixed up shortly

**Comment by askvortsov1** [comment]
@datitisev done, sorry for the delay

**Pull request merged**